go/ct/evm_fuzz_test.go

```diff
@@ -0,0 +1,285 @@
+			t.Fatal("invalid result, resulting state does not match reference state:", testeeResultState.Diff(referenceResultState), errorReportString(state, testeeResultState, referenceResultState))
+		}
+	})
+}
+
+func fuzzVm(testee ct.Evm, f *testing.F) {
+
+	rnd := rand.New(0)
+
+	prepareFuzzingSeeds(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
+		state, err := corpusInputDataToCtState(opCodes, gas, revision, stackBytes)
+		if err != nil {
+			t.Skip(err)
+		}
+
+		result, _ := testee.StepN(state, 1)
+		result.Release()
+	})
+}
+
+const (
+	fuzzIdealStackSize     int = 7  // < max pops in a single instruction
+	fuzzMaximumCodeSegment int = 33 // < 1 instruction with 32 data bytes
+)
+
+// prepareFuzzingSeeds is a helper function to be used by similar fuzzing tests
+// the arguments passed to the f.Add function needs to match the arguments
+// passed to the f.Fuzz function in type, position and number
+func prepareFuzzingSeeds(f *testing.F, rnd *rand.Rand) {
+
+	// every possible revision
+	for revision := MinRevision; revision <= NewestSupportedRevision; revision++ {
+		// every possible opCode, even if invalid
+		for op := 0x00; op <= 0xFF; op++ {
+			// Some gas values: this is a hand made sampling of interesting values,
+			// the fuzzer will generate more interesting values around these, the initial
+			// list just sketches a region of interest around which the fuzzer will generate
+			// more values. I found no measurable difference about being more accurate.
+			for _, gas := range []int64{0, 1, 6, 10, 1000, math.MaxInt64} {
+
+				// generate a code segment with the operation followed by 6 random values
+				ops := make([]byte, fuzzMaximumCodeSegment)
+				rand.Read(ops[:])
+				ops[0] = byte(op)
+
+				// generate a stack: the stack contains a mixture of values
+				// as seed for mutations
+				stack := make([]byte, fuzzIdealStackSize*32)
+				// fill a quarter with random values
+				var i int
+				for i = 0; i < 7*8; i++ {
+					stack[i] = byte(rnd.Int31n(256))
+				}
+				// leave a quarter with zeros
+				for ; i < 7*16; i++ {
+					stack[i] = 0
+				}
+				// fill a quarter with max values
+				for ; i < 7*24; i++ {
+					stack[i] = 255
+				}
+				// fill a quarter with avg values
+				for ; i < 7*32; i++ {
+					stack[i] = 127
+				}
+
+				f.Add(
+					ops,            // opCodes
+					int64(gas),     // gas
+					byte(revision), // revision
+					stack,          // stack
+				)
+			}
+		}
+	}
+
+	// add one more with a full stack
+	ops := make([]byte, fuzzMaximumCodeSegment)
+	rnd.Read(ops[:])
+	ops[0] = byte(0x00)
+	fullStack := make([]byte, 1024*32)
+	rnd.Read(fullStack[:])
+	f.Add(
+		ops,                // opCodes
+		int64(0),           // gas
+		byte(R07_Istanbul), // revision
+		fullStack,          // stack
+	)
+}
+
+func corpusInputDataToCtState(opCodes []byte, gas int64, revision byte, stackBytes []byte) (*st.State, error) {
```

Comment:
This logic of this function might be complex enough to justify a unit test to make sure it is working as intended.
The test could be added in this file.